SwiftReflector/VTableDetails.cs

```diff
@@ -0,0 +1,490 @@
+
+			var invocation = new CSFunctionCall ($"{localVTName}.{FuncName ()}!", false, args.ToArray ());
+
+			var callLine = delType.Type == CSSimpleType.Void ? new CSLine (invocation) :
+				CSReturn.ReturnLine (invocation);
+			body.Add (callLine);
+			use.AddIfNotPresent (typeof (UnmanagedCallersOnlyAttribute));
+			var attr = CSAttribute.FromAttr (typeof (UnmanagedCallersOnlyAttribute), new CSArgumentList (), true);
+			attr.AttachBefore (unmanagedReceiver);
+			UnmanagedReceivers.Methods.Add (unmanagedReceiver);
+		}
+
+		public CSMethod ImplementVirtualPropertyStaticReceiver (CSType thisType, string csProxyName, CSDelegateTypeDecl delType,
+					       CSUsingPackages use, FunctionDeclaration funcDecl, CSProperty prop, CSMethod protoListMethod,
+					       TypeMapper typeMapper, bool isObjC, bool hasAssociatedTypes, Func<int, int, string> genericRenamer = null)
+		{
+			var returnType = funcDecl.IsGetter ? delType.Type : CSSimpleType.Void;
+
+			var pl = delType.Parameters;
+			var plWithoutGCPtr = new CSParameterList (pl.GetRange (1, pl.Count - 1));
+			var usedIDs = new List<string> ();
+			usedIDs.AddRange (pl.Select (p => p.Name.Name));
+
+			var body = new CSCodeBlock ();
+
+			if (protoListMethod != null) {
+				body.Add (CSFunctionCall.FunctionCallLine ("throw new NotImplementedException", false, CSConstant.Val ($"Property method {protoListMethod.Name.Name} protocol list type is not supported yet")));
+			} else {
+				var marshaler = new MarshalEngineCSafeSwiftToCSharp (use, usedIDs, typeMapper);
+				marshaler.GenericRenamer = genericRenamer;
+
+				var bodyContents = marshaler.MarshalFromLambdaReceiverToCSProp (prop, thisType, csProxyName,
+												delType.Parameters,
+												funcDecl, prop.PropType, isObjC, hasAssociatedTypes);
+				body.AddRange (bodyContents);
+			}
+
+			var methodKind = CSMethodKind.Static;
+			if (delType.IsUnsafe || ParameterListContainsPointers (pl))
+				methodKind = CSMethodKind.StaticUnsafe;
+
+			var recvr = new CSMethod (CSVisibility.None, methodKind, returnType, new CSIdentifier (RecevierName ()), plWithoutGCPtr, body);
+
+			// write unmanaged receiver
+			ImplementUnmanagedReceiver (methodKind, delType, pl, plWithoutGCPtr, usedIDs, use);
+
+			currentVTEntry++;
+			return recvr;
+		}
+
+		public CSMethod ImplementVirtualSubscriptStaticReceiver (CSType thisType, string csProxyName, CSDelegateTypeDecl delType, CSUsingPackages use,
+						FunctionDeclaration funcDecl, CSProperty prop, CSMethod protoListMethod, TypeMapper typeMapper, bool isObjC,
+						Func<int, int, string> genericRenamer = null, bool hasAssociatedTypes = false)
+		{
+			var returnType = funcDecl.IsSubscriptGetter ? delType.Type : CSSimpleType.Void;
+
+			var pl = delType.Parameters;
+			var plWithoutGCPtr = new CSParameterList (pl.Skip (1));
+			var usedIDs = new List<string> ();
+			usedIDs.AddRange (pl.Select (p => p.Name.Name));
```

Comment:
The List<T> class already has a constructor that takes a constructor that does the init and adds the range: https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.-ctor?view=net-8.0#system-collections-generic-list-1-ctor(system-collections-generic-ienumerable((-0)))